### PR TITLE
Remove manual pkgconfig provides from non-toolchain specs

### PR DESCRIPTION
**Pull request by oliviacrain**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

In Mariner 2.0, manual `Provides: pkgconfig(*)` statements are no longer necessary. These provides are generated automatically by RPM. The effect of these manual provides is benign when written properly with proper versions in the proper subpackages, but can cause build tooling confusion when done incorrectly. So, we should remove the manual provides statements.

This work was completed for the toolchain in #2155. This PR extends that work to non-toolchain specs.

###### Change Log  

- Remove manual `pkgconfig(*)` provides from non-toolchain specs.

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Pipeline build looks good, generated pkgconfig provides look good in changed specs

### CODE REVIEW

File: SPECS/libxkbfile/libxkbfile.spec [hunk 1]
```diff
@@ -21,8 +21,6 @@ Summary:        X.Org X11 libxkbfile development package
 
 Requires:       %{name} = %{version}-%{release}
 
-Provides:       pkgconfig(xkbfile) = %{version}-%{release}
```
Comment (PawelWMS) [inline SPECS/libxkbfile/libxkbfile.spec:24]: Missed `Release` update.

**Review by PawelWMS — COMMENTED** [review]
Looks good, just two small comments.

File: SPECS/libXdamage/libXdamage.spec [hunk 2]
```diff
@@ -24,8 +24,6 @@ Summary:        Development files for %{name}
 
 Requires:       %{name} = %{version}-%{release}
 
-Provides:       pkgconfig(xdamage) = %{version}-%{release}
```
[the remark this answers is not in the document]
Comment (oliviacrain) [reply SPECS/libXdamage/libXdamage.spec:27]: Thanks for catching this, fixed!

File: SPECS/libxkbfile/libxkbfile.spec (hunk 1 above)
Comment (oliviacrain) [reply SPECS/libxkbfile/libxkbfile.spec:24]: Thanks for catching this, fixed!

**Review by PawelWMS — APPROVED** [review]

**Comment by dmcilvaney** [comment]
Could you double check that the tooling updates I did are still happy? (Check the end of the package build phase, where it lists all the .rpm files it built. If there are any conflicts it will list any conflicting .rpms in red).

I don't think there should be any issues though.

**Review by dmcilvaney — APPROVED** [review]

**Pull request merged**